### PR TITLE
Fix/condo/doma 3099/added filtering processing tasks on frontend

### PR DESCRIPTION
**Pull request by AntonAL**
Filtering on backend is not enough because for User with `isAdmin` flag we need to:
- return all tasks in Keystone admin section
- return only his own tasks in "public" section

## Review note
First commit is a refactoring.
The objective of this PR is in d81d9c1be7044f3cea8a0e91f7a409d5c76be160 commit

### CODE REVIEW

File: apps/condo/domains/common/components/tasks/index.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import React from 'react'
+import { User } from '../../../../schema'
```
Comment (sitozzz) [inline apps/condo/domains/common/components/tasks/index.tsx:2]: absolute import please

File: apps/condo/domains/common/components/tasks/storage/TasksLocalStorage.tsx [hunk 2]
```diff
@@ -23,16 +23,19 @@ const isServerSide = typeof window === 'undefined'
...
         }
+        if (!user) {
+            return { records: [] }
+        }
         const tasks = this.getAllItemsFromStorage()
         if (!status) {
             return tasks
         }
-        const records = tasks.filter(task => task.status === status)
+        const records = tasks.filter(task => task.status === status && task.user && task.user.id === user.id)
```
Comment (sitozzz) [inline apps/condo/domains/common/components/tasks/storage/TasksLocalStorage.tsx:38]:
```suggestion
        const records = tasks.filter(task => task.status === status && get(task, 'user.id') === user.id)
```

**Review by sitozzz — COMMENTED** [review]
Need to test it locally again with miniapp flow

File: apps/condo/domains/common/components/tasks/index.tsx (hunk 1 above)
Comment (AleX83Xpert) [reply apps/condo/domains/common/components/tasks/index.tsx:2]:
```suggestion
import { User } from '@app/condo/schema'
```

**Review by AleX83Xpert — APPROVED** [review]

**Review by sitozzz — CHANGES_REQUESTED** [review]
There is some issues with auth. You need to handle this cases

Screenshot 2022-08-04 at 12 10 14

Screenshot 2022-08-04 at 11 22 19

**Comment by AntonAL** [comment]
Resolved the issues "Cannot read properties" and other.
They are related to wrong hooks usage.

**Review by sitozzz — APPROVED** [review]
Now it's ok

**Pull request merged**